### ISSUE TITLE
aws_vpc_peering_connection fails to manage "requester" options when using peering across accounts

### ISSUE THREAD

**Issue by cornfeedhobo** [terraform-providers/terraform-provider-aws#3676]
### Terraform Version
Terraform v0.11.3

### Affected Resource(s)
- aws_vpc_peering_connection

### Terraform Configuration Files
```hcl
resource "aws_vpc_peering_connection" "requester" {
  provider = "aws.requester"

  count = "${local.enabled}"
  tags  = "${merge(local.tags, map("Side", "Requestor"))}"

  peer_owner_id = "${var.accepter-account_id}"
  peer_vpc_id   = "${var.accepter-vpc_id}"
  vpc_id        = "${var.requester-vpc_id}"

  requester {
    allow_remote_vpc_dns_resolution = "true" // this isn't allowed using cross-account?
  }

  accepter {
    allow_remote_vpc_dns_resolution = "true" // this is ignored when not using auto-accept?
  }
}

resource "aws_vpc_peering_connection_accepter" "accepter" {
  provider = "aws.accepter"

  count = "${local.enabled}"
  tags  = "${merge(local.tags, map("Side", "Acceptor"))}"

  vpc_peering_connection_id = "${aws_vpc_peering_connection.requester.id}"
  auto_accept               = "true"
}
```

### Expected Behavior
The `requester` block should apply, but there is no mechanism to supply different credentials for just that operation.

### Actual Behavior
```
* aws_vpc_peering_connection.requester: Error modifying VPC Peering Connection options: OperationNotPermitted: User 012345678901 does not have permission to modify the accepter side peering options
```

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
This seems to only relate to peering VPCs in different accounts.

### Suggested solution
If the AST has a `aws_vpc_peering_connection_accepter`, use it, and it's provider, to apply the configuration block.

**Comment by ajcann** [comment]
I've also just run into this issue.

**Comment by ewbankkit** [comment]
I think this is the same problem with the [`allow_remote_vpc_dns_resolution`](https://www.terraform.io/docs/providers/aws/r/vpc_peering.html#allow_remote_vpc_dns_resolution) attribute as in
* https://github.com/terraform-providers/terraform-provider-aws/issues/3850
* https://github.com/terraform-providers/terraform-provider-aws/issues/3069

See also https://github.com/terraform-providers/terraform-provider-aws/pull/3097.

**Comment by ewbankkit** [comment]
See my comments [here](https://github.com/terraform-providers/terraform-provider-aws/issues/3069#issuecomment-375800030) that attempt to explain why this can't be resolved without a new `aws_vpc_peering_connection_options` resource.

**Comment by bflad** [comment]
The new `aws_vpc_peering_connection_options` resource for managing these dependency issues has been merged via #3909 and will release with v1.17.0 of the AWS provider, likely in two days.

**Comment by bflad** [comment]
The new resource has been released in [version 1.17.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1170-may-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
